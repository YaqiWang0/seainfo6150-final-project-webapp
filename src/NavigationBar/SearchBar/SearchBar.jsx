import React, { useEffect, useState} from "react"
import  style from "./SearchBar.module.css"
import { useHistory, useLocation } from "react-router-dom";

const SearchBar = (props) => {
    const [redirect, setRedirect] = useState(false);
    const [data, setData] = useState();

    let history = useHistory();
    let location = useLocation();

    console.log(location.pathname)
    const handleSubmit = e => {
        e.preventDefault();
        console.log(location.pathname)
        history.push(
            "/search",
            {
                data: e.target.elements.search.value,
                login: props.login,
                credential : props.credential
            }
        )
    }

    return (
        <div>
            <form action="" className= {style.searchBar} onSubmit={handleSubmit}>
                <input type="search" name="search" pattern=".*\S.*" required/>
                    <button className={style.searchBtn} type="submit">
                        <span>Search</span>
                    </button>
            </form>
       </div>
    )
}

export default SearchBar