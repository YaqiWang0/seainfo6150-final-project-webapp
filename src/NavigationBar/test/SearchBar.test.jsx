
import React from 'react'
import SearchBar from "../SearchBar/SearchBar";


describe("SearchBar test", () => {

    it('render successfully', function () {
        const {container} = render(
            <SearchBar/>
        );
        expect(container).toMatchSnapshot();
    })
})