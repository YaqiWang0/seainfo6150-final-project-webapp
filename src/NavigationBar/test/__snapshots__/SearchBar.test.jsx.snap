// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar test render successfully 1`] = `
<div>
  <div>
    <form
      action=""
      class="searchBar"
    >
      <input
        name="search"
        pattern=".*\\\\S.*"
        required=""
        type="search"
      />
      <button
        class="searchBtn"
        type="submit"
      >
        <span>
          Search
        </span>
      </button>
    </form>
  </div>
</div>
`;
